File: Source/Extensions/Tessa.Extensions.Default.Client/Workflow/KrProcess/KrGlobalTileContainer.cs
using System.Collections.Generic;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.Platform;

namespace Tessa.Extensions.Default.Client.Workflow.KrProcess
{
    /// <inheritdoc cref="IKrGlobalTileContainer"/>
    public sealed class KrGlobalTileContainer :
        IKrGlobalTileContainer
    {
        #region Fields

        private volatile IReadOnlyList<KrTileInfo> infos;

        #endregion

        #region IKrGlobalTileContainer Members

        /// <inheritdoc />
        public void Init(
            IReadOnlyList<KrTileInfo> globalTiles)
        {
            Check.ArgumentNotNull(globalTiles, nameof(globalTiles));

            this.infos = globalTiles;
        }

        /// <inheritdoc />
        public IReadOnlyList<KrTileInfo> GetTileInfos() => this.infos;

        #endregion
    }
}

File: WebClient SDK/src/default/workflow/krProcess/krGlobalTileContainer.ts
import { KrTileInfo } from 'tessa/workflow/krProcess';

export class KrGlobalTileContainer {

  //#region ctor

  private constructor() {}

  //#endregion

  //#region instance

  private static _instance: KrGlobalTileContainer;

  public static get instance(): KrGlobalTileContainer {
    if (!KrGlobalTileContainer._instance) {
      KrGlobalTileContainer._instance = new KrGlobalTileContainer();
    }
    return KrGlobalTileContainer._instance;
  }

  //#endregion

  //#region fields

  private _infos: KrTileInfo[];

  //#endregion

  //#region methods

  public init(infos: KrTileInfo[]) {
    this._infos = infos;
  }

  public getTileInfos(): ReadonlyArray<KrTileInfo> {
    return this._infos || [];
  }

  //#endregion

}